niri-config/src/lib.rs

```diff
@@ -1401,6 +1440,55 @@ impl CornerRadius {
     }
 }
 
+impl FromStr for GradientInterpolation {
+    type Err = miette::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut iter = s.split_whitespace();
+        let in_part1 = iter.next();
+        let in_part2 = iter.next();
+
+        let color = if in_part1 != None {
+            let in_str = in_part1.unwrap();  
+            match in_str {
+                "srgb" => GradientColorSpace::Srgb,
+                "srgb-linear" => GradientColorSpace::SrgbLinear,
+                "display-p3" => GradientColorSpace::DisplayP3,
+                "a98-rgb" => GradientColorSpace::A98Rgb,
+                "prophoto-rgb" => GradientColorSpace::ProphotoRgb,
+                "rec2020" => GradientColorSpace::Rec2020,
+                "lab" => GradientColorSpace::Lab,
+                "oklab" => GradientColorSpace::Oklab,
+                "xyz" => GradientColorSpace::Xyz,
+                "xyz-d50" => GradientColorSpace::XyzD50,
+                "xyz-d65" => GradientColorSpace::XyzD65,
+                "hsl" => GradientColorSpace::Hsl,
+                "hwb" => GradientColorSpace::Hwb,
+                "lch" => GradientColorSpace::Lch,
+                "oklch" => GradientColorSpace::Oklch,
+                &_ => return Err(miette!("Invalid color-space: {in_str}"))
+            }
+        } else {
+            GradientColorSpace::Srgb
+        };
+
+        let interpolation = if in_part2 != None {
```

Comment:
Should this also check that a polar space was parsed on the step above? Because this only makes sense for polar spaces

Reply:
i should probably implement that for user simplicity however on the rendering end it makes little difference as the hue interpolation type is ignored for non polar spaces